src/Exception/TransferException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client\Exception;
+
+/**
+ * Thrown when an error related to sending/receiving the request/response occurs
```

Comment:
"Base exception for all php-http exceptions."

basically its either RequestException or BatchException.